src/cheri_insts.sail

```diff
@@ -1346,10 +1346,27 @@ function clause execute (LoadDataCap(rd, cs1, is_unsigned, width)) = {
   handle_load_data_via_cap(rd, 0b0 @ cs1, cs1_val, vaddr, is_unsigned, width)
 }
 
+union load_cap_ptw_ext_result = {
+  LCPE_OK : Capability,
+  LCPE_Exn : ExceptionType
+}
+
+val handle_load_cap_ptw_ext : (Capability, bool, ext_ptw_lc) -> load_cap_ptw_ext_result
+function handle_load_cap_ptw_ext (v, vialc, ptw_info) =
```

Comment:
I'd prefer this function be inlined; it's only used twice (maybe 3 times with AMOs) so the code size is likely going to be no different, and it avoids having more magic helpers. I feel currently this just makes the code harder to read, especially with the Permit_Load_Cap handling being pushed down into here.